#!/bin/bash
#
# Call with following arguments
# bash RNASeq_STAR.sh <Input_Reads1.fq.gz> <Input_Reads2.fq.gz> <output_basename>
# 

#Absolute path of fastq files
fwd=$1
rev=$2

#Basename of output files
bn=$3

#Path to reference genome and Index files.
star_ref="/home/yi/hg19/STAR_Index"

#STAR 2 pass basic mode run
echo -e "["$(date)"]\tAligning.."

nice -n 10 STAR --runThreadN 15 --quantMode GeneCounts --twopassMode Basic --alignSoftClipAtReferenceEnds No --chimSegmentMin 20 --chimJunctionOverhangMin 20 --chimScoreMin 1 --outFilterIntronMotifs RemoveNoncanonical --outFilterType BySJout --outSAMtype BAM SortedByCoordinate --genomeDir $star_ref --readFilesCommand zcat --readFilesIn $fwd $rev --outFileNamePrefix $bn"_"

samtools index ${bn}_Aligned.sortedByCoord.out.bam

# For unstranded RNA-seq data, Cufflinks/Cuffdiff requires spliced alignments with XS strand attribute, which could be generated by --outSAMstrandField intronMotif option

echo -e "["$(date)"]\tDONE!"
